tests/test_pool.py

```diff
@@ -9,12 +9,13 @@
 @pytest.mark.parametrize("max_size", [50, 10, 5, 1])
 async def test_simple(max_size, loop):
     counter = 0
+    lock = asyncio.Lock()
 
     async def create_instance():
         nonlocal counter
-        await asyncio.sleep(0)
-        counter += 1
-        return counter
+        async with lock:
```

Comment:
the main idea here is this block might be completely lock-free in the user code.

Reply:
I removed the lock and made the time while holding and instance conditional on whether `max_size` has been reached or not. The code will sleep longer it has not been and sleeps shorter when it has been.

(My PC only got up to 3 instances when it was sleeping 0.01 seconds.)